### PR TITLE
Add a first-class provider definition and fix a few bugs.

### PR DESCRIPTION
**Pull request by pgavlin**
- Fix type definitions and status codes in the OpenFaaS client
- Do not return an error from `Read` for a missing resource
- Support skipping TLS certificate verification for e.g. interacting
  with installs that use self-signed certs

### CODE REVIEW

File: sdk/nodejs/provider.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export class Provider extends pulumi.ProviderResource {
+    /**
+     * Create a provider resource with the given unique name, arguments, and options.
+     *
+     * @param name The _unique_ name of the resource.
+     * @param args The arguments to use to populate this provider's configuration.
+     * @param opts A bag of options that control this resource's behavior.
+     */
+    constructor(name: string, args: ProviderArgs, opts?: pulumi.ResourceOptions) {
+        super("pulumi:providers:openfaas", name, {
```
Comment (lukehoban) [inline sdk/nodejs/provider.ts:15]: Is this the naming pattern we use for other providers?  (putting them in `pulumi:providers`?)

**Review by lukehoban — APPROVED** [review]
LGTM

File: sdk/nodejs/provider.ts (hunk 1 above)
Comment (pgavlin) [reply sdk/nodejs/provider.ts:15]: Yes. This pattern is in fact a requirement, as it is how the engine recognizes a provider resource.

Comment (lukehoban) [reply sdk/nodejs/provider.ts:15]: :+1:

**Pull request merged**